### PR TITLE
Added registration, logging in, JWTs

### LINKED ISSUE #13: Registration and logging in
New users should be able to register an account and existing users should be able to log in to their accounts. They can either do this by visiting the My Profile/Login page in the navigation bar, or when they attempt to perform a user-only action':
- Liking a post
- Reposting a post
- Following a profile

### PR DESCRIPTION
**Pull request by melleg**
Closes #13 
Added registration as well as JWT-based logging in authentication.

**Notes**
You need to declare a JWT_SECRET system environment variable of sufficient length to start the server.

**Still to do:**
Making the frontend actually refresh their JWT if it has expired,

### CODE REVIEW

File: frontend/src/auth/auth-service.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import RegisterModel from "./register-model";
+
+const uri = "auth";
+
+export const login = async (model: LoginModel) => {
+  return (await api.post<string>(`${uri}/login`, model)).data;
+};
+
+export const register = async (model: RegisterModel) => {
+  return (await api.post(`${uri}/logi`, model)).data;
```
Comment (mio-xenoblade) [inline frontend/src/auth/auth-service.ts]: logi doesnt work, replace with register

Comment (mio-xenoblade) [reply frontend/src/auth/auth-service.ts]: when replaced with register, and registering an account, attempting to login with the new account results in a "java.lang.IllegalArgumentException: There is no PasswordEncoder mapped for the id "null"" error in backend (status 500)

File: frontend/src/auth/auth-service.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { api } from "../base-api";
+import LoginModel from "./login-model";
+import RegisterModel from "./register-model";
+
+const uri = "auth";
+
+export const login = async (model: LoginModel) => {
+  return (await api.post<string>(`${uri}/login`, model)).data;
```
Comment (mio-xenoblade) [inline frontend/src/auth/auth-service.ts:8]: attempting to log in with a non-seeder account results in a password encoder id "null" error

**Review by mio-xenoblade — CHANGES_REQUESTED** [review]

**Review by mio-xenoblade — APPROVED** [review]
maybe change register user to prevent use of spaces in username (replace with - or something)

**Comment by melleg** [comment]
Good point, will make that a new issue

> maybe change register user to prevent use of spaces in username (replace with - or something)

**Pull request merged**